packages/openapi-react-query/tsconfig.json

```
@@ -0,0 +1,15 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "lib": ["ESNext", "DOM"],
+    "module": "ES2022",
+    "moduleResolution": "Node",
```

```suggestion
    "moduleResolution": "Bundler",
```

This should be preferred for React/frontend projects

---

🤔 Oh you know what? `openapi-fetch` has this too; that should be updated.

---

Openapi-fetch is isomorphic and it doesn't use a bundler to publish so it should probably use nodenext instead of bundler for moduleresolution

---

True. A React project like this should be `Bundler` for sure; openapi-fetch is debatable and could be `Bundler` or `NodeNext` because the latter is not a predictable environment for frontend (but AFAIK wouldn’t cause issues).

---

Why wouldn't it be a predictable environment for frontend? AFAIK this setting only controls the module environment, so how modules should be interpreted and outputted by ts. Since nodenext is a more restricted module setting than bundler, it would seem to me that it should be the much more predictable setting for both backend and frontend projects. Bundler allows additionally for all sort of ambiguity like extensionless module import paths which then need to be modelled correctly by the build tool the user of this lib will be using.

This is now definitely off topic for this PR. So feel free to go ahead. But still worthwhile to think about. :)

---

I do not know enough about this property to have a good opinion on that. So let's debate about this for a future PR.